### ISSUE TITLE
Bindings - Code examples improperly marked

### ISSUE THREAD

**Issue by wesbland** [mpi-forum/mpi-issues#602]
# Problem

Code examples should be marked so that the code checker can confirm that they at least compile - the information on this is in the instructions, and there are a zillion example in the document. The code checker has found many problems and helps ensure a higher quality document. I noted problems in binding-2.tex, but will need to review all of the other examples now.
In that chapter, many examples are marked with `%%SKIP` when they should be properly marked for the language etc.

# Proposal

Follow the instructions in `instr.tex/pdf`. If there are problems or questions, contact @wgropp.

# Changes to the Text

No changes to the text.  The code checker uses structured comments.

# Impact on Implementations

None.

# Impact on Users

Improved quality of the document.

# References

This is covered in instr.pdf.

### Instructions

If there is nothing to do in your chapter, just comment and close this issue. Otherwise, create a PR to address the problems.

**Comment by VictorEijkhout** [comment]
I'm starting to dig into this. What is the code checker tool? As in specifically what program do I compile/run?

**Comment by wgropp** [comment]
It is a little kludgy, but "make check" runs the tests.  There is a configure in mpicompilechk that tries to set things up. The C tests have a special mpi.h file; the Fortran tests, unfortunately, currently rely on an MPI's implementation Fortran and the mpi or mpi_f08 module.

**Comment by VictorEijkhout** [comment]
```
[Current/istc/mpi-standard 1053] make check
make: *** No rule to make target `mpicompilechk/mpichk', needed by `check'.  Stop.
```

There is a file `mpichk.in`. Copying that one is not enough:

```
Checking examples in ./chap-ei
/bin/sh: ./mpicompilechk/mpichk: @PERL5@: bad interpreter: No such file or directory
```

It seems to need some localization step.

**Comment by VictorEijkhout** [comment]
```
Once the source files have been annotated, the command mpichk should be used
to extract and compile the example code.  This command is created by configure,
as in

   ./configure
```
Sorry for the noise. First run `autoconf` which btw complains about obsolete macros.

**Comment by cniethammer** [comment]
@VictorEijkhout if the configure script is missing for you in the `mpicompilechk` directory you can generate the `./configure` running `autoreconf -vf` therein. Afterwards you can run therein make.

Then you can call mpichk from the top directory for your sub chapter by e.g.
```
./mpicompilechk/mpichk --outdir=chap-ei chap-ei/ei-2.tex 
Opening file chap-ei/ei-2.tex
```

**Comment by wgropp** [comment]
I'll improve the make process to ensure that the prerequisites are built or information provided

**Comment by VictorEijkhout** [comment]
@cniethammer 
 
> autoreconf -vf

```
[istc/mpi-standard/mpicompilechk 1064] autoreconf -vf
autoreconf: export WARNINGS=
autoreconf: warning: autoconf input should be named 'configure.ac', not 'configure.in'
```

Is my autoconf too old or should that file be renamed?

**Comment by wgropp** [comment]
No, this is a gratuitous change that was made in autoconf - you can ignore this.  At some point, I'll move the file to change the name. The configure.in file was written when the input was required to be configure.in ; configure.ac would not have been correct at the time.

Another reminder that computing no longer considers backward compatibility to be a positive property.

**Comment by wgropp** [comment]
I've renamed configure.in to configure.ac in mpicompilechk as part of some updates.
